File: pinyougou-parent/pinyougou-web-manager/src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){

    //读取用户列表
    this.findAll = function(){
        return $http.get('../user/findAll.do');
    }


    //分页
    this.findPage=function(page,rows){
        return $http.get('../user/findPage.do?page='+page+'&rows='+rows);
    }

    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../user/search.do?page='+page+"&rows="+rows, searchEntity);
    }

    //冻结
    this.updateStatus = function (ids,status) {
        return $http.get('../user/updateStatus.do?ids='+ids+'&status='+status);
    }
})

File: pinyougou-parent/pinyougou-interface/src/main/java/cn/itcast/core/service/UserService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.user.User;
import entity.PageResult;

import java.util.List;

public interface UserService {
    void sendCode(String phone);

    void add(User user, String smscode);

    //查询所有用户
    List<User> findAll();

    //添加查询
    PageResult search(Integer pageNum, Integer pageSize, User user);

    //更新状态
    void updateStatus(Long[] ids, String status);
    //完善用户信息
    void perfectionMessage(User user);
    //回显用户信息
    User findVO(String name);
}
